cmd/juju/commands/bootstrap.go

```diff
@@ -354,6 +357,13 @@ func (c *bootstrapCommand) Init(args []string) (err error) {
 		return errors.New("--db-snap-asserts requires --db-snap")
 	}
 
+	if c.ControllerCharmPath != "" {
+		_, err := c.Filesystem().Stat(c.ControllerCharmPath)
```

Comment:
Is this where we could do a bit of additional validation on the charm file, e.g., ensure `ReadCharmArchive()` returns success?

Reply:
Yup, added extra checks